tests/TestCase/Routing/RouterTest.php

```diff
@@ -3087,6 +3090,103 @@ public function testScope()
+            'prefix' => 'myapp_cake_someprefix_' . $tmpName,
+            'path' => CACHE . 'persistent/',
+        ]);
+
+        Router::scope('/path', ['param' => 'value', '_cache' => 'routes'], function ($routes) {
+            $this->assertInstanceOf('Cake\Routing\RouteBuilder', $routes);
+            $this->assertEquals('/path', $routes->path());
+            $this->assertEquals(['param' => 'value'], $routes->params());
+            $this->assertEquals('', $routes->namePrefix());
+
+            $routes->connect('/articles', ['controller' => 'Articles']);
+        });
+
+        $this->assertFileExists($routePath);
+        $serializedCollection = Cache::read('routes' . '/path', 'routes');
+        $expected = [
+            'pass' => [],
+            'controller' => 'Articles',
+            'action' => 'index',
+            'param' => 'value',
+            'plugin' => null,
+            '_matchedRoute' => '/path/articles'
+        ];
+        $this->assertInstanceOf('\Cake\Routing\RouteCollection', $serializedCollection);
+        $this->assertSame($expected, $serializedCollection->parse('/path/articles'));
+
+        return $routePath;
+    }
+
+    /**
+     * Test read cached scope
+     * @depends testCreateCacheScope
+     * @return void
+     */
+    public function testReadCacheScope($routePath = '')
+    {
+        $this->assertFileExists($routePath);
+
+        Router::scope('/path', ['param' => 'value', '_cache' => 'routes'], function ($routes) {
+            $this->fail('Cached routes should have been used');
+        });
+
+        $expected = [
+            'pass' => [],
+            'controller' => 'Articles',
+            'action' => 'index',
+            'param' => 'value',
+            'plugin' => null,
+            '_matchedRoute' => '/path/articles'
+        ];
+        $this->assertSame($expected, Router::parseRequest(new ServerRequest('/path/articles')));
+        Cache::drop('routes');
+        $file = new File($routePath);
+        $file->delete();
+    }
+
+    /**
+     * Test cached scope in a non serialized cache engine
+     *
+     * @return void
+     */
+    public function testCacheScopeNotSerialized()
+    {
+        $routePath = tempnam(CACHE . 'persistent/', 'myapp_cake_someprefix_');
+        $tmpName = substr($routePath, strlen(CACHE . 'persistent/' . 'myapp_cake_someprefix_'));
+
+        Cache::setConfig('routes', [
+            'className' => 'File',
+            'prefix' => 'myapp_cake_someprefix_' . $tmpName,
+            'path' => CACHE . 'persistent/',
+            'serialize' => false,
+        ]);
+
+        Router::scope('/path', ['param' => 'value', '_cache' => 'routes'], function ($routes) {
+            $this->assertInstanceOf('Cake\Routing\RouteBuilder', $routes);
+            $this->assertEquals('/path', $routes->path());
+            $this->assertEquals(['param' => 'value'], $routes->params());
+            $this->assertEquals('', $routes->namePrefix());
```

Comment:
Can we use assertSame()? Equals would also return true for 0 or other falsy content here.